velox/common/memory/ByteStream.h

```diff
@@ -404,11 +412,17 @@ class AppendWindow {
   AppendWindow(ByteOutputStream& stream, Scratch& scratch)
       : stream_(stream), scratchPtr_(scratch) {}
 
-  ~AppendWindow() {
+  ~AppendWindow() noexcept {
```

Comment:
We probably doesn't need noexcept annotation. I guess a VELOX_FAIL in catch section with a clear error message is good enough? Thanks!